backend/src/routes/api/integrations/nim/index.ts

```diff
@@ -61,36 +60,45 @@ module.exports = async (fastify: KubeFastifyInstance) => {
         reply: FastifyReply,
       ) => {
         const enableValues = request.body;
-
-        await createNIMSecret(fastify, namespace, enableValues)
-          .then(async () => {
-            await createNIMAccount(fastify, namespace)
-              .then((response) => {
-                const isEnabled = isAppEnabled(response);
-                reply.send({
-                  isInstalled: true,
-                  isEnabled: isEnabled,
-                  canInstall: false,
-                  error: '',
-                });
-              })
-              .catch((e) => {
-                const message = `Failed to create NIM account, ${e.response?.body?.message}`;
-                fastify.log.error(message);
-                reply.status(e.response.statusCode).send(new Error(message));
+        try {
+          const { createdNew } = await createNIMSecret(fastify, enableValues);
+          if (createdNew) {
+            const response = await createNIMAccount(fastify);
+            const isEnabled = isAppEnabled(response);
+            reply.send({
+              isInstalled: true,
+              isEnabled: isEnabled,
+              canInstall: false,
+              error: '',
+            });
+          } else {
+            // Secret was updated, so we skip creating the NIM account but still need to get the status.
+            const account = await getNIMAccount(fastify);
+            if (account) {
+              const isEnabled = isAppEnabled(account);
+              reply.send({
+                isInstalled: true,
+                isEnabled: isEnabled,
+                canInstall: false,
+                error: '',
               });
-          })
-          .catch((e) => {
-            if (e.response?.statusCode === 409) {
-              fastify.log.error(`NIM secret already exists, skipping creation.`);
-              reply.status(409).send(new Error(`NIM secret already exists, skipping creation.`));
             } else {
-              fastify.log.error(`Failed to create NIM secret. ${e.response?.body?.message}`);
-              reply
-                .status(e.response.statusCode)
-                .send(new Error(`Failed to create NIM secret, ${e.response?.body?.message}`));
+              // Account not found after the secret was updated
+              fastify.log.error(`NIM account not found after updating secret.`);
+              reply.status(404).send(new Error('NIM account not found.'));
             }
-          });
+          }
+        } catch (e: any) {
+          if (e.response?.statusCode === 409) {
+            fastify.log.error(`NIM secret already exists, skipping creation.`);
+            reply.status(409).send(new Error(`NIM secret already exists, skipping creation.`));
```

Comment:
You note:
```
            // Secret was updated, so we skip creating the NIM account but still need to get the status.
```
Is this even possible anymore?

Reply:
So the `e.response?.statusCode === 409` is still possible ..but from the `const response = await createNIMAccount(fastify);` because there is no longer the `.catch` for it. So if `createNIMAccount(fastify)` does fail the error message would be logged as the "Nim secret already exists,..." when it should be "Nim account already exists..."

So we also need to make sure NIMAccount api calls are still being caught properly

Reply:
With the latest update that checks if the account and secret exist, the 409 in here should no longer be possible again I believe?

Reply:
Yes, you are right.